rasa/nlu/selectors/response_selector.py

```diff
@@ -439,6 +440,12 @@ def process(self, message: Message, **kwargs: Any) -> None:
 
         self._set_message_property(message, prediction_dict, selector_key)
 
+        if out and DIAGNOSTIC_DATA in out:
```

Comment:
why do we need to check if `DIAGNOSTIC_DATA` is in `out`?

Reply:
Otherwise we'd call `values_to_numpy` with `None` and if we let that return `None` we'd still call `message.add_diagnostic_data` with `None` and it looks like we always add something there, which I'd find confusing.